libvast/src/system/index.cpp

```diff
@@ -1237,32 +1284,81 @@ index(index_actor::stateful_pointer<index_state> self,
                   }
                   // Adjust layout stats by subtracting the events of the
                   // removed partition.
-                  const auto* partition = fbs::GetPartition(chunk->data());
-                  if (partition->partition_type()
-                      != fbs::partition::Partition::legacy) {
-                    rp.deliver(caf::make_error(ec::format_error, "unexpected "
-                                                                 "format "
-                                                                 "version"));
-                    return;
-                  }
-                  vast::ids all_ids;
-                  const auto* partition_legacy
-                    = partition->partition_as_legacy();
-                  for (const auto* partition_stats :
-                       *partition_legacy->type_ids()) {
-                    const auto* name = partition_stats->name();
-                    vast::ids ids;
-                    if (auto error
-                        = fbs::deserialize_bytes(partition_stats->ids(), ids)) {
-                      rp.deliver(caf::make_error(ec::format_error,
-                                                 "could not deserialize "
-                                                 "ids: "
-                                                   + render(error)));
+                  if (chunk->size() >= FLATBUFFERS_MAX_BUFFER_SIZE) {
+                    VAST_WARN("failed to load partition for deletion at {} "
+                              "because its size of {} exceeds the maximum "
+                              "allowed size of {}. The index statistics will "
+                              "be incorrect until the database has been "
+                              "rebuilt and restarted",
+                              path, chunk->size(), FLATBUFFERS_MAX_BUFFER_SIZE);
+                    // !!! Fallback path. We try to erase the corresponding
+                    // segment store file if it exists.
+                    // TODO: Also attempt to erase parquet stores once that
+                    // implementation is merged.
```

Comment:
It's merged now, so probably this needs to become a discussion story about extending the store plugin API to allow for lookup by partition id without having a store header available.

Reply:
Oh dear, I hoped I could avoid this for the scope of this PR. Given that oversized partitions should be much rarer from this point forward I think we should keep this PR on segments only.

Reply:
Ideally "much rarer" will be "never" :D I agree on keeping this PR on segments, but let's at least reword the TODO